src/App.tsx

```diff
@@ -1,17 +1,126 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import cn from 'classnames';
 
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { PreparedTodo } from './app.typedef';
+
+const preparedTodos: PreparedTodo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [todosList, setTodosList] = useState([...preparedTodos]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleInputError, setTitleInputError] = useState(false);
+  const [userInputError, setUserInputError] = useState(false);
+
+  const addtodo = (title: string, userId: number) => {
+    const newTodo: PreparedTodo = {
+      userId,
+      id: todosList.length + 1,
```

Comment:
almost everyone has this one

> - `id` is a maximal `id` in the array + 1

not `list.length`
something like this:
`Math.max(...array.map((todo) => todo.id)) + 1`